kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -221,7 +223,8 @@ private TransactionCommitResult doCommit(
     }
     setTxnOpt.ifPresent(setTxn -> metadataActions.add(createTxnSingleAction(setTxn.toRow())));
 
-    try (CloseableIterator<Row> stageDataIter = dataActions.iterator()) {
+    try (CloseableIterator<Row> stageDataIter =
+        new ValidateDomainMetadataIterator(protocol, dataActions.iterator(), FULL_SCHEMA)) {
```

Comment:
I see. So this is just a thin wrapper over the actions-to-be-committed and validates the DomainMetadata on write.

@vkorukanti -- would we ever want to validate other things on write?

@qiyuandong-db -- I wonder if we should not have a `ValidateDomainMetadataIterator` but rather a `ValidationIterator`. This validation iterator lets you pass in different validationFunctions that take a row and decide to or not to throw an error. This would be a more extensible solution.

For example, I'd be fine removing the validation from this PR and coding that ^ up in a followup PR ... let's see what @vkorukanti thinks

Reply:
I’ve updated this part to separate the domain metadata actions from `dataActions`. Now, they’re stored in a `List<domainMetadata>` and we can apply validation functions directly. So the `ValidateDomainMetadataIterator` is no longer needed here.

We can implement a general `ValidationIterator` in a separate PR if the need comes.